src/Orchard.Web/Modules/Orchard.Taxonomies/Services/TaxonomyService.cs

```diff
@@ -54,13 +57,21 @@ public class TaxonomyService : ITaxonomyService {
 
             Logger = NullLogger.Instance;
             T = NullLocalizer.Instance;
+
+            // initialize memorization structures
+            _taxonomiesByName = new Dictionary<string, TaxonomyPart>();
+            _taxonomiesBySlug = new Dictionary<string, TaxonomyPart>();
         }
 
         public ILogger Logger { get; set; }
         public Localizer T { get; set; }
 
+        private IEnumerable<TaxonomyPart> _taxonomies;
```

Comment:
```suggestion
        private List<TaxonomyPart> _taxonomies;
```

Since the type is known and this is a private member

Reply:
Unless it's not what the method returns

Reply:
`IContentQuery<T>.List()` (which is what's assigned and eventually returned here) returns an `IEnumerable<T>`. Definitely not a `List<T>`.